### PR TITLE
Add initial take at api-based tech test

### PR DESCRIPTION
**Pull request by michalboo**
This circulated as a google doc and was used on some people already (I believe).
I re-worked it a little bit and added it to the repo.

### CODE REVIEW

File: CTM-SDET-test-API.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* Search for users by email
+
+1. Please write automated tests that cover some of the mentioned functionality. They should serve as an example of the kind of test you would find useful given the above context.
+1. If time permits, please add tests for some edge cases or negative scenarios that you can think of.
+
+## Languages
+
+You may use any programming language of your choice. We would prefer that you use a language that will help you show off your skills in building test automation frameworks, rather than one that you are unfamiliar with.
+
+The main languages that we use at CTM for test automation are Javascript, C# and Ruby.
```
Comment (zecarrera) [inline CTM-SDET-test-API.md]: In the UI tech test I rewrote this sentence as:
Most of the existing test frameworks at CTM are using javascript, but we also have some legacy ones written in ruby and C#.
Don't know if you agree, but I believe it would be more up to date with our current reality?

File: CTM-SDET-test-API.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  * Email
+  * Address
+  * Phone
+  * Website
+* Create a new user
+* Update user details
+* Search for users by email
+
+1. Please write automated tests that cover some of the mentioned functionality. They should serve as an example of the kind of test you would find useful given the above context.
+1. If time permits, please add tests for some edge cases or negative scenarios that you can think of.
```
Comment (zecarrera) [inline CTM-SDET-test-API.md]: 2 instead of 1?

Comment (michalboo) [reply CTM-SDET-test-API.md]: Markdown, so it renders correctly (odd, I know – I reckon it's a way to have an easier way of managing long ordered lists)

Comment (michalboo) [reply CTM-SDET-test-API.md]: Having said that, this isn't a long ordered list, and it's likely people will be viewing it in plaintext, will fix.

File: CTM-SDET-test-API.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  * Email
+  * Address
+  * Phone
+  * Website
+* Create a new user
+* Update user details
+* Search for users by email
+
+1. Please write automated tests that cover some of the mentioned functionality. They should serve as an example of the kind of test you would find useful given the above context.
+2. If time permits, please add tests for some edge cases or negative scenarios that you can think of.
```
[the remark this answers is not in the document]
Comment (michalboo) [reply CTM-SDET-test-API.md]: On reflection, I can't think of a nice way to frame it as part of a tech test.
I figure asking questions along the lines of: "How would you explore the functionality of an API?" or "How would you translate requirements to test cases?" …could be a worthwhile step to follow up with, but asking candidates to come up with their own requirements, after pointing them at a URL, seems like it would mostly confuse things. (Happy to capture this in a follow up update)

File: CTM-SDET-test-API.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Most of the existing test frameworks at CTM are using Javascript, but we also have some legacy ones written in Ruby and C#.
+
+## Documentation
+
+You should provide instructions on how to run your tests, and note any prerequisites (eg. operating system, language). Include these in a README. If you wish to highlight anything in your tech test, you should do so in the README. The ease with which the interviewers can get your tests running is an important part of the test.
+
+## Submission
+
+Create a public repository on GitHub with your code and send us the link.
```
[the remark this answers is not in the document]
Comment (michalboo) [reply CTM-SDET-test-API.md]: I don't think that's a huge problem, to be honest. But will add your suggestion as an alternative 👍
